File: ruby/render_file.rb
require "docxmerge"

docxmerge = Docxmerge::Docxmerge.new("API_KEY", "default", "https://api.docxmerge.com")

response = docxmerge.render_file(
    File.open("../fixtures/hello_world.docx", "r"),
    {
        :logo => "https://docxmerge.com/assets/android-chrome-512x512.png",
        :name => "James Bond",
    },
    "PDF"
)

file_output = "./hello_world_ruby_file.pdf"

f = File.open(file_output, "wb")
f << response
f.close
puts "Check #{file_output}"

File: php/render_file.php
<?php
require_once('vendor/autoload.php');

use Docxmerge\Docxmerge;

$docxmerge = new Docxmerge("API_KEY");

$fp = fopen("./hello_world_php.pdf", "w");

$docxmerge->renderFile($fp, "../fixtures/hello_world.docx", array(
    "name" => "James bond",
    "logo" => "https://docxmerge.com/assets/android-chrome-512x512.png"
), "PDF");
